tensorflow/python/ops/nn_ops.py

```diff
@@ -149,14 +149,12 @@ def __init__(
                                                               conv_dims))
     if conv_dims == 1:
       # conv1d uses the 2-d data format names
-      if data_format is None or data_format == "NWC":
-        data_format_2d = "NHWC"
-      elif data_format == "NCW":
-        data_format_2d = "NCHW"
-      else:
+      if data_format is None:
+        data_format = "NWC"
+      elif data_format not in {"NCW", "NWC"}:
```

Comment:
Before it only gave a deprecation warning if the user passed "NCHW" or "NHWC" but now it raises an error. Change {"NCW", "NWC"} to {"NCW", "NWC", "NCHW", "CHWC"}